### ISSUE TITLE
cloudflare_zone_settings_override not working for TF 0.12

### ISSUE THREAD

**Issue by georgekaz** [cloudflare/cf-terraforming#158]
after export and immediate plan you get:

```
Error: Missing required argument

  on zone.tf line 542, in resource "cloudflare_zone_settings_override" "zone_settings_override_XXX":
 542: resource "cloudflare_zone_settings_override" "zone_settings_override_XXX" {

The argument "zone_id" is required, but no definition was found.


Error: Unsupported argument

  on zone.tf line 543, in resource "cloudflare_zone_settings_override" "zone_settings_override_XXX":
 543:   name = "XXX"

An argument named "name" is not expected here.

Error: Unsupported argument

  on zone.tf line 545, in resource "cloudflare_zone_settings_override" "zone_settings_override_XXXX":
 545:     advanced_ddos            = "on"

An argument named "advanced_ddos" is not expected here.


Error: Unsupported argument

  on zone.tf line 554, in resource "cloudflare_zone_settings_override" "zone_settings_override_XXX":
 554:     ciphers                  = []

An argument named "ciphers" is not expected here.
```

**Comment by thedotedge** [comment]
Same issue with 0.13, after exporting and running `terraform plan`:

* The argument "zone_id" is required, but no definition was found.
* An argument named "edge_cache_ttl" is not expected here.
* An argument named "ciphers" is not expected here.
* An argument named "advanced_ddos" is not expected here.

**Comment by gstotts** [comment]
@georgekaz In regards to the first error, this was a change that was made to the terraform provider as it moved to version 2.  I'm no familiar with the cf-terraforming tool to know if this was addressed in it but if you replace the zone (zone name) with zone_id referencing the ID of the zone, it should eliminate the error.   See [here](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/guides/version-2-upgrade#zone-name-to-zone-id-changes) for more specifics on changes to the provider that occurred in the move to v2.

**Comment by gstotts** [comment]
#182

**Comment by georgekaz** [comment]
thanks @gstotts . I had no issues editing the tf to make it work, I was just reporting these as cf-terraforming problems with the latest provider.

**Comment by Anna-Katona** [comment]
Are there any plans to fix it (I mean remove advanced_ddos, ciphers, etc)? 
For now, we are using a custom script to rewrite the output according to the new provider but it would be great to have the proper output from the beginning?

It would be also great to post the actual terraform and cloudflare provider versions in README for example

**Comment by patryk** [comment]
Thank you for patience everyone and sorry for delay.

- The argument "zone_id" is required, but no definition was found.
Fixed in #182
- An argument named "edge_cache_ttl" is not expected here.
Fixed in 483460097e6c10fe9ae295bc6cb4fc045bf6ea56
- An argument named "ciphers" is not expected here.
Fixed in d9c9ae4d819eb201a3370907d96dcb37b58d0bb0
- An argument named "advanced_ddos" is not expected here.
Fixed in 483460097e6c10fe9ae295bc6cb4fc045bf6ea56

**Comment by georgekaz** [comment]
thanks @patryk and for fixing https://github.com/cloudflare/cf-terraforming/issues/157. This tool was extremely useful for me.
